### PR TITLE
statistics: use analysis worker to run analysis jobs

### PR DESCRIPTION
**Pull request by Rustin170506**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/55618

close https://github.com/pingcap/tidb/pull/55621

Problem Summary:

### What changed and how does it work?

Started using analysis worker to run analysis jobs. The analysis jobs are always spawned whenever possible.

1. Init the analysis worker when creating the refresher.
2. Only create one single refresher per stats handle.

For unit tests, I inject WaitAutoAnalyzeFinishedForTest to wait for all analysis jobs to finish. This would help us make the tests easier because many tests depend on this API to analyze tables.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
	- [x] [partitioned tables](https://github.com/pingcap/tidb/pull/55848#issuecomment-2330758088) 
	- [x] [multiple tables](https://github.com/pingcap/tidb/pull/55848#issuecomment-2330820456)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by Rustin170506** [comment]
Tested locally:
No verbose logs and it works correctly:
```log
[2024/09/05 14:59:13.342 +08:00] [INFO] [advancer.go:524] ["Subscription handler spawned."] [category="log backup subscription manager"]
[2024/09/05 14:59:13.359 +08:00] [INFO] [domain.go:3155] ["instance plan eviction"] [detail="memory usage is below the soft limit, currentTot: 0, softLimit: 104857600"] [num_evicted=0] [time_spent=1.542µs]
[2024/09/05 14:59:22.365 +08:00] [INFO] [refresher.go:126] ["Auto analyze triggered"] [category=stats] [job="DynamicPartitionedTableAnalysisJob:\n\tAnalyzeType: analyzeDynamicPartitionIndex\n\tPartitions: p13, p18, p20, p5, p9, p10, p12, p16, p4, p3, p6, p7, p17, p1, p8, p11, p14, p15, p19, p2\n\tPartitionIndexes: map[PRIMARY:[p5 p13 p18 p20 p4 p9 p10 p12 p16 p1 p3 p6 p7 p17 p2 p8 p11 p14 p15 p19] idx_column1:[p4 p9 p10 p12 p16 p1 p3 p6 p7 p17 p2 p8 p11 p14 p15 p19 p5 p13 p18 p20]]\n\tSchema: test\n\tGlobal Table: t\n\tGlobal TableID: 104\n\tTableStatsVer: 2\n\tChangePercentage: 1.000000\n\tTableSize: 18000.00\n\tLastAnalysisDuration: 30m0s\n\tWeight: 3.368389\n"]
[2024/09/05 14:59:22.365 +08:00] [INFO] [worker.go:91] ["Job submitted"] [category=stats] [job="DynamicPartitionedTableAnalysisJob:\n\tAnalyzeType: analyzeDynamicPartitionIndex\n\tPartitions: p13, p18, p20, p5, p9, p10, p12, p16, p4, p3, p6, p7, p17, p1, p8, p11, p14, p15, p19, p2\n\tPartitionIndexes: map[PRIMARY:[p5 p13 p18 p20 p4 p9 p10 p12 p16 p1 p3 p6 p7 p17 p2 p8 p11 p14 p15 p19] idx_column1:[p4 p9 p10 p12 p16 p1 p3 p6 p7 p17 p2 p8 p11 p14 p15 p19 p5 p13 p18 p20]]\n\tSchema: test\n\tGlobal Table: t\n\tGlobal TableID: 104\n\tTableStatsVer: 2\n\tChangePercentage: 1.000000\n\tTableSize: 18000.00\n\tLastAnalysisDuration: 30m0s\n\tWeight: 3.368389\n"]
[2024/09/05 14:59:22.412 +08:00] [INFO] [analyze.go:391] ["save analyze results concurrently"] [buildStatsConcurrency=1] [saveStatsConcurrency=2]
[2024/09/05 14:59:22.436 +08:00] [INFO] [save.go:200] ["incrementally update modifyCount"] [category=stats] [tableID=110] [curModifyCnt=3000] [results.BaseModifyCnt=3000] [modifyCount=0]
[2024/09/05 14:59:22.436 +08:00] [INFO] [save.go:222] ["directly update count"] [category=stats] [tableID=110] [results.Count=3000] [count=3000]
[2024/09/05 14:59:22.445 +08:00] [INFO] [save.go:200] ["incrementally update modifyCount"] [category=stats] [tableID=118] [curModifyCnt=3000] [results.BaseModifyCnt=3000] [modifyCount=0]
[2024/09/05 14:59:22.445 +08:00] [INFO] [save.go:222] ["directly update count"] [category=stats] [tableID=118] [results.Count=3000] [count=3000]
[2024/09/05 14:59:22.515 +08:00] [INFO] [analyze.go:567] ["analyze table `test`.`t` has finished"] [partition=p5] ["job info"="auto analyze table all indexes, columns id, partition_key, column1 with 256 buckets, 100 topn, 1 samplerate"] ["start time"=2024/09/05 14:59:22.427 +08:00] ["end time"=2024/09/05 14:59:22.507 +08:00] [cost=79.72725ms] ["sample rate reason"="use min(1, 110000/3000) as the sample-rate=1"]
[2024/09/05 14:59:22.516 +08:00] [INFO] [save.go:200] ["incrementally update modifyCount"] [category=stats] [tableID=123] [curModifyCnt=3000] [results.BaseModifyCnt=3000] [modifyCount=0]
[2024/09/05 14:59:22.516 +08:00] [INFO] [save.go:222] ["directly update count"] [category=stats] [tableID=123] [results.Count=3000] [count=3000]
[2024/09/05 14:59:22.528 +08:00] [INFO] [analyze.go:567] ["analyze table `test`.`t` has finished"] [partition=p13] ["job info"="auto analyze table all indexes, columns id, partition_key, column1 with 256 buckets, 100 topn, 1 samplerate"] ["start time"=2024/09/05 14:59:22.436 +08:00] ["end time"=2024/09/05 14:59:22.522 +08:00] [cost=86.451667ms] ["sample rate reason"="use min(1, 110000/3000) as the sample-rate=1"]
[2024/09/05 14:59:22.529 +08:00] [INFO] [save.go:200] ["incrementally update modifyCount"] [category=stats] [tableID=125] [curModifyCnt=3000] [results.BaseModifyCnt=3000] [modifyCount=0]
[2024/09/05 14:59:22.529 +08:00] [INFO] [save.go:222] ["directly update count"] [category=stats] [tableID=125] [results.Count=3000] [count=3000]
[2024/09/05 14:59:22.594 +08:00] [INFO] [analyze.go:567] ["analyze table `test`.`t` has finished"] [partition=p18] ["job info"="auto analyze table all indexes, columns id, partition_key, column1 with 256 buckets, 100 topn, 1 samplerate"] ["start time"=2024/09/05 14:59:22.444 +08:00] ["end time"=2024/09/05 14:59:22.588 +08:00] [cost=144.037583ms] ["sample rate reason"="use min(1, 110000/3000) as the sample-rate=1"]
[2024/09/05 14:59:22.596 +08:00] [INFO] [save.go:200] ["incrementally update modifyCount"] [category=stats] [tableID=109] [curModifyCnt=3000] [results.BaseModifyCnt=3000] [modifyCount=0]
[2024/09/05 14:59:22.596 +08:00] [INFO] [save.go:222] ["directly update count"] [category=stats] [tableID=109] [results.Count=3000] [count=3000]
[2024/09/05 14:59:22.608 +08:00] [INFO] [analyze.go:567] ["analyze table `test`.`t` has finished"] [partition=p20] ["job info"="auto analyze table all indexes, columns id, partition_key, column1 with 256 buckets, 100 topn, 1 samplerate"] ["start time"=2024/09/05 14:59:22.452 +08:00] ["end time"=2024/09/05 14:59:22.602 +08:00] [cost=149.124625ms] ["sample rate reason"="use min(1, 110000/3000) as the sample-rate=1"]
[2024/09/05 14:59:22.609 +08:00] [INFO] [save.go:200] ["incrementally update modifyCount"] [category=stats] [tableID=114] [curModifyCnt=3000] [results.BaseModifyCnt=3000] [modifyCount=0]
[2024/09/05 14:59:22.609 +08:00] [INFO] [save.go:222] ["directly update count"] [category=stats] [tableID=114] [results.Count=3000] [count=3000]
[2024/09/05 14:59:22.671 +08:00] [INFO] [analyze.go:567] ["analyze table `test`.`t` has finished"] [partition=p4] ["job info"="auto analyze table all indexes, columns id, partition_key, column1 with 256 buckets, 100 topn, 1 samplerate"] ["start time"=2024/09/05 14:59:22.461 +08:00] ["end time"=2024/09/05 14:59:22.666 +08:00] [cost=205.351416ms] ["sample rate reason"="use min(1, 110000/3000) as the sample-rate=1"]
[2024/09/05 14:59:22.671 +08:00] [INFO] [save.go:200] ["incrementally update modifyCount"] [category=stats] [tableID=115] [curModifyCnt=3000] [results.BaseModifyCnt=3000] [modifyCount=0]
[2024/09/05 14:59:22.671 +08:00] [INFO] [save.go:222] ["directly update count"] [category=stats] [tableID=115] [results.Count=3000] [count=3000]
[2024/09/05 14:59:22.685 +08:00] [INFO] [analyze.go:567] ["analyze table `test`.`t` has finished"] [partition=p9] ["job info"="auto analyze table all indexes, columns id, partition_key, column1 with 256 buckets, 100 topn, 1 samplerate"] ["start time"=2024/09/05 14:59:22.469 +08:00] ["end time"=2024/09/05 14:59:22.680 +08:00] [cost=210.422083ms] ["sample rate reason"="use min(1, 110000/3000) as the sample-rate=1"]
[2024/09/05 14:59:22.686 +08:00] [INFO] [save.go:200] ["incrementally update modifyCount"] [category=stats] [tableID=117] [curModifyCnt=3000] [results.BaseModifyCnt=3000] [modifyCount=0]
[2024/09/05 14:59:22.686 +08:00] [INFO] [save.go:222] ["directly update count"] [category=stats] [tableID=117] [results.Count=3000] [count=3000]
[2024/09/05 14:59:22.771 +08:00] [INFO] [analyze.go:567] ["analyze table `test`.`t` has finished"] [partition=p10] ["job info"="auto analyze table all indexes, columns id, partition_key, column1 with 256 buckets, 100 topn, 1 samplerate"] ["start time"=2024/09/05 14:59:22.477 +08:00] ["end time"=2024/09/05 14:59:22.764 +08:00] [cost=286.522916ms] ["sample rate reason"="use min(1, 110000/3000) as the sample-rate=1"]
[2024/09/05 14:59:22.772 +08:00] [INFO] [save.go:200] ["incrementally update modifyCount"] [category=stats] [tableID=121] [curModifyCnt=3000] [results.BaseModifyCnt=3000] [modifyCount=0]
[2024/09/05 14:59:22.772 +08:00] [INFO] [save.go:222] ["directly update count"] [category=stats] [tableID=121] [results.Count=3000] [count=3000]
[2024/09/05 14:59:22.796 +08:00] [INFO] [analyze.go:567] ["analyze table `test`.`t` has finished"] [partition=p12] ["job info"="auto analyze table all indexes, columns id, partition_key, column1 with 256 buckets, 100 topn, 1 samplerate"] ["start time"=2024/09/05 14:59:22.515 +08:00] ["end time"=2024/09/05 14:59:22.785 +08:00] [cost=270.062333ms] ["sample rate reason"="use min(1, 110000/3000) as the sample-rate=1"]
[2024/09/05 14:59:22.797 +08:00] [INFO] [save.go:200] ["incrementally update modifyCount"] [category=stats] [tableID=106] [curModifyCnt=3000] [results.BaseModifyCnt=3000] [modifyCount=0]
[2024/09/05 14:59:22.797 +08:00] [INFO] [save.go:222] ["directly update count"] [category=stats] [tableID=106] [results.Count=3000] [count=3000]
[2024/09/05 14:59:22.893 +08:00] [INFO] [analyze.go:567] ["analyze table `test`.`t` has finished"] [partition=p16] ["job info"="auto analyze table all indexes, columns id, partition_key, column1 with 256 buckets, 100 topn, 1 samplerate"] ["start time"=2024/09/05 14:59:22.529 +08:00] ["end time"=2024/09/05 14:59:22.888 +08:00] [cost=359.402292ms] ["sample rate reason"="use min(1, 110000/3000) as the sample-rate=1"]
[2024/09/05 14:59:22.893 +08:00] [INFO] [save.go:200] ["incrementally update modifyCount"] [category=stats] [tableID=108] [curModifyCnt=3000] [results.BaseModifyCnt=3000] [modifyCount=0]
[2024/09/05 14:59:22.893 +08:00] [INFO] [save.go:222] ["directly update count"] [category=stats] [tableID=108] [results.Count=3000] [count=3000]
[2024/09/05 14:59:22.908 +08:00] [INFO] [analyze.go:567] ["analyze table `test`.`t` has finished"] [partition=p1] ["job info"="auto analyze table all indexes, columns id, partition_key, column1 with 256 buckets, 100 topn, 1 samplerate"] ["start time"=2024/09/05 14:59:22.595 +08:00] ["end time"=2024/09/05 14:59:22.903 +08:00] [cost=308.721583ms] ["sample rate reason"="use min(1, 110000/3000) as the sample-rate=1"]
[2024/09/05 14:59:22.909 +08:00] [INFO] [save.go:200] ["incrementally update modifyCount"] [category=stats] [tableID=111] [curModifyCnt=3000] [results.BaseModifyCnt=3000] [modifyCount=0]
[2024/09/05 14:59:22.909 +08:00] [INFO] [save.go:222] ["directly update count"] [category=stats] [tableID=111] [results.Count=3000] [count=3000]
[2024/09/05 14:59:22.975 +08:00] [INFO] [analyze.go:567] ["analyze table `test`.`t` has finished"] [partition=p3] ["job info"="auto analyze table all indexes, columns id, partition_key, column1 with 256 buckets, 100 topn, 1 samplerate"] ["start time"=2024/09/05 14:59:22.608 +08:00] ["end time"=2024/09/05 14:59:22.970 +08:00] [cost=361.486083ms] ["sample rate reason"="use min(1, 110000/3000) as the sample-rate=1"]
[2024/09/05 14:59:22.975 +08:00] [INFO] [save.go:200] ["incrementally update modifyCount"] [category=stats] [tableID=112] [curModifyCnt=3000] [results.BaseModifyCnt=3000] [modifyCount=0]
[2024/09/05 14:59:22.975 +08:00] [INFO] [save.go:222] ["directly update count"] [category=stats] [tableID=112] [results.Count=3000] [count=3000]
[2024/09/05 14:59:22.990 +08:00] [INFO] [analyze.go:567] ["analyze table `test`.`t` has finished"] [partition=p6] ["job info"="auto analyze table all indexes, columns id, partition_key, column1 with 256 buckets, 100 topn, 1 samplerate"] ["start time"=2024/09/05 14:59:22.671 +08:00] ["end time"=2024/09/05 14:59:22.985 +08:00] [cost=314.023166ms] ["sample rate reason"="use min(1, 110000/3000) as the sample-rate=1"]
[2024/09/05 14:59:22.991 +08:00] [INFO] [save.go:200] ["incrementally update modifyCount"] [category=stats] [tableID=122] [curModifyCnt=3000] [results.BaseModifyCnt=3000] [modifyCount=0]
[2024/09/05 14:59:22.991 +08:00] [INFO] [save.go:222] ["directly update count"] [category=stats] [tableID=122] [results.Count=3000] [count=3000]
[2024/09/05 14:59:23.061 +08:00] [INFO] [analyze.go:567] ["analyze table `test`.`t` has finished"] [partition=p7] ["job info"="auto analyze table all indexes, columns id, partition_key, column1 with 256 buckets, 100 topn, 1 samplerate"] ["start time"=2024/09/05 14:59:22.685 +08:00] ["end time"=2024/09/05 14:59:23.053 +08:00] [cost=367.790833ms] ["sample rate reason"="use min(1, 110000/3000) as the sample-rate=1"]
[2024/09/05 14:59:23.062 +08:00] [INFO] [save.go:200] ["incrementally update modifyCount"] [category=stats] [tableID=107] [curModifyCnt=3000] [results.BaseModifyCnt=3000] [modifyCount=0]
[2024/09/05 14:59:23.062 +08:00] [INFO] [save.go:222] ["directly update count"] [category=stats] [tableID=107] [results.Count=3000] [count=3000]
[2024/09/05 14:59:23.085 +08:00] [INFO] [analyze.go:567] ["analyze table `test`.`t` has finished"] [partition=p17] ["job info"="auto analyze table all indexes, columns id, partition_key, column1 with 256 buckets, 100 topn, 1 samplerate"] ["start time"=2024/09/05 14:59:22.771 +08:00] ["end time"=2024/09/05 14:59:23.075 +08:00] [cost=304.319834ms] ["sample rate reason"="use min(1, 110000/3000) as the sample-rate=1"]
[2024/09/05 14:59:23.086 +08:00] [INFO] [save.go:200] ["incrementally update modifyCount"] [category=stats] [tableID=113] [curModifyCnt=3000] [results.BaseModifyCnt=3000] [modifyCount=0]
[2024/09/05 14:59:23.086 +08:00] [INFO] [save.go:222] ["directly update count"] [category=stats] [tableID=113] [results.Count=3000] [count=3000]
[2024/09/05 14:59:23.158 +08:00] [INFO] [analyze.go:567] ["analyze table `test`.`t` has finished"] [partition=p2] ["job info"="auto analyze table all indexes, columns id, partition_key, column1 with 256 buckets, 100 topn, 1 samplerate"] ["start time"=2024/09/05 14:59:22.797 +08:00] ["end time"=2024/09/05 14:59:23.153 +08:00] [cost=356.58325ms] ["sample rate reason"="use min(1, 110000/3000) as the sample-rate=1"]
[2024/09/05 14:59:23.158 +08:00] [INFO] [save.go:200] ["incrementally update modifyCount"] [category=stats] [tableID=116] [curModifyCnt=3000] [results.BaseModifyCnt=3000] [modifyCount=0]
[2024/09/05 14:59:23.158 +08:00] [INFO] [save.go:222] ["directly update count"] [category=stats] [tableID=116] [results.Count=3000] [count=3000]
[2024/09/05 14:59:23.175 +08:00] [INFO] [analyze.go:567] ["analyze table `test`.`t` has finished"] [partition=p8] ["job info"="auto analyze table all indexes, columns id, partition_key, column1 with 256 buckets, 100 topn, 1 samplerate"] ["start time"=2024/09/05 14:59:22.893 +08:00] ["end time"=2024/09/05 14:59:23.170 +08:00] [cost=276.882375ms] ["sample rate reason"="use min(1, 110000/3000) as the sample-rate=1"]
[2024/09/05 14:59:23.175 +08:00] [INFO] [save.go:200] ["incrementally update modifyCount"] [category=stats] [tableID=119] [curModifyCnt=3000] [results.BaseModifyCnt=3000] [modifyCount=0]
[2024/09/05 14:59:23.176 +08:00] [INFO] [save.go:222] ["directly update count"] [category=stats] [tableID=119] [results.Count=3000] [count=3000]
[2024/09/05 14:59:23.242 +08:00] [INFO] [analyze.go:567] ["analyze table `test`.`t` has finished"] [partition=p11] ["job info"="auto analyze table all indexes, columns id, partition_key, column1 with 256 buckets, 100 topn, 1 samplerate"] ["start time"=2024/09/05 14:59:22.909 +08:00] ["end time"=2024/09/05 14:59:23.238 +08:00] [cost=329.032625ms] ["sample rate reason"="use min(1, 110000/3000) as the sample-rate=1"]
[2024/09/05 14:59:23.243 +08:00] [INFO] [save.go:200] ["incrementally update modifyCount"] [category=stats] [tableID=120] [curModifyCnt=3000] [results.BaseModifyCnt=3000] [modifyCount=0]
[2024/09/05 14:59:23.243 +08:00] [INFO] [save.go:222] ["directly update count"] [category=stats] [tableID=120] [results.Count=3000] [count=3000]
[2024/09/05 14:59:23.255 +08:00] [INFO] [analyze.go:567] ["analyze table `test`.`t` has finished"] [partition=p14] ["job info"="auto analyze table all indexes, columns id, partition_key, column1 with 256 buckets, 100 topn, 1 samplerate"] ["start time"=2024/09/05 14:59:22.975 +08:00] ["end time"=2024/09/05 14:59:23.251 +08:00] [cost=276.566667ms] ["sample rate reason"="use min(1, 110000/3000) as the sample-rate=1"]
[2024/09/05 14:59:23.256 +08:00] [INFO] [save.go:200] ["incrementally update modifyCount"] [category=stats] [tableID=124] [curModifyCnt=3000] [results.BaseModifyCnt=3000] [modifyCount=0]
[2024/09/05 14:59:23.256 +08:00] [INFO] [save.go:222] ["directly update count"] [category=stats] [tableID=124] [results.Count=3000] [count=3000]
[2024/09/05 14:59:23.311 +08:00] [INFO] [analyze.go:567] ["analyze table `test`.`t` has finished"] [partition=p15] ["job info"="auto analyze table all indexes, columns id, partition_key, column1 with 256 buckets, 100 topn, 1 samplerate"] ["start time"=2024/09/05 14:59:22.990 +08:00] ["end time"=2024/09/05 14:59:23.308 +08:00] [cost=318.01ms] ["sample rate reason"="use min(1, 110000/3000) as the sample-rate=1"]
[2024/09/05 14:59:23.324 +08:00] [INFO] [analyze.go:567] ["analyze table `test`.`t` has finished"] [partition=p19] ["job info"="auto analyze table all indexes, columns id, partition_key, column1 with 256 buckets, 100 topn, 1 samplerate"] ["start time"=2024/09/05 14:59:23.061 +08:00] ["end time"=2024/09/05 14:59:23.319 +08:00] [cost=257.453792ms] ["sample rate reason"="use min(1, 110000/3000) as the sample-rate=1"]
[2024/09/05 14:59:23.327 +08:00] [INFO] [global_stats.go:106] ["use async merge global stats"] [category=stats] [sampled=] [tableID=104] [table=t]
[2024/09/05 14:59:43.359 +08:00] [INFO] [domain.go:3155] ["instance plan eviction"] [detail="memory usage is below the soft limit, currentTot: 0, softLimit: 104857600"] [num_evicted=0] [time_spent=1.5µs]
[2024/09/05 15:00:13.358 +08:00] [INFO] [domain.go:3155] ["instance plan eviction"] [detail="memory usage is below the soft limit, currentTot: 0, softLimit: 104857600"] [num_evicted=0] [time_spent=1.833µs]
[2024/09/05 15:00:43.358 +08:00] [INFO] [domain.go:3155] ["instance plan eviction"] [detail="memory usage is below the soft limit, currentTot: 0, softLimit: 104857600"] [num_evicted=0] [time_spent=1.958µs]
```

**Comment by Rustin170506** [comment]
Tested with multiple tables:
1. Start the cluster with binary: `tiup playground nightly --db.binpath /Users/rustin/code/tidb/bin/tidb-server`
2. Run insert data script:
```rust
#!/usr/bin/env -S cargo +nightly -Zscript
---cargo
[dependencies]
clap = { version = "4.2", features = ["derive"] }
sqlx = { version = "0.7", features = ["runtime-tokio-rustls", "mysql"] }
tokio = { version = "1", features = ["full"] }
fake = { version = "2.5", features = ["derive"] }
---

use clap::Parser;
use fake::{Fake, Faker};
use sqlx::mysql::MySqlPoolOptions;

#[derive(Parser, Debug)]
#[clap(version)]
struct Args {
    #[clap(short, long, help = "MySQL connection string")]
    database_url: String,
}

#[derive(Debug)]
struct TableRow {
    id: i64,
    column1: String,
    column2: i32,
    column3: i32,
    column4: String,
}

#[tokio::main]
async fn main() -> Result<(), sqlx::Error> {
    let args = Args::parse();

    let pool = MySqlPoolOptions::new()
        .max_connections(5)
        .connect(&args.database_url)
        .await?;

    // Create 20 tables
    for i in 0..20 {
        let table_name = format!("t{}", i);
        let create_table_query = format!(
            "CREATE TABLE IF NOT EXISTS {} (
                id BIGINT NOT NULL PRIMARY KEY,
                column1 VARCHAR(255) NOT NULL,
                column2 INT NOT NULL,
                column3 INT NOT NULL,
                column4 VARCHAR(255) NOT NULL,
                INDEX idx_column1 (column1)
            )",
            table_name
        );

        sqlx::query(&create_table_query)
            .execute(&pool)
            .await?;

        println!("Created table: {}", table_name);

        // Insert 3000 rows into each table
        for _ in 0..3000 {
            let row = TableRow {
                id: Faker.fake::<i64>(),
                column1: Faker.fake::<String>(),
                column2: Faker.fake::<i32>(),
                column3: Faker.fake::<i32>(),
                column4: Faker.fake::<String>(),
            };

            let insert_query = format!(
                "INSERT INTO {} (id, column1, column2, column3, column4)
                VALUES (?, ?, ?, ?, ?)",
                table_name
            );

            sqlx::query(&insert_query)
                .bind(row.id)
                .bind(&row.column1)
                .bind(row.column2)
                .bind(row.column3)
                .bind(&row.column4)
                .execute(&pool)
                .await?;
        }

        println!("Successfully inserted 3000 rows into table '{}'.", table_name);
    }

    Ok(())
}
```
3. Check analysis records:

| id | update\_time | table\_schema | table\_name | partition\_name | job\_info | processed\_rows | start\_time | end\_time | state | fail\_reason | instance | process\_id |
| :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- |
| 1 | 2024-09-05 15:32:13 | test | t0 |  | auto analyze table all indexes, columns id, column1 with 256 buckets, 100 topn, 1 samplerate | 3000 | 2024-09-05 15:32:13 | 2024-09-05 15:32:13 | finished | null | 127.0.0.1:4000 | null |
| 2 | 2024-09-05 15:32:13 | test | t19 |  | auto analyze table all indexes, columns id, column1 with 256 buckets, 100 topn, 1 samplerate | 3000 | 2024-09-05 15:32:13 | 2024-09-05 15:32:13 | finished | null | 127.0.0.1:4000 | null |
| 3 | 2024-09-05 15:32:16 | test | t1 |  | auto analyze table all indexes, columns id, column1 with 256 buckets, 100 topn, 1 samplerate | 3000 | 2024-09-05 15:32:16 | 2024-09-05 15:32:16 | finished | null | 127.0.0.1:4000 | null |
| 4 | 2024-09-05 15:32:16 | test | t18 |  | auto analyze table all indexes, columns id, column1 with 256 buckets, 100 topn, 1 samplerate | 3000 | 2024-09-05 15:32:16 | 2024-09-05 15:32:16 | finished | null | 127.0.0.1:4000 | null |
| 5 | 2024-09-05 15:32:19 | test | t2 |  | auto analyze table all indexes, columns id, column1 with 256 buckets, 100 topn, 1 samplerate | 3000 | 2024-09-05 15:32:19 | 2024-09-05 15:32:19 | finished | null | 127.0.0.1:4000 | null |
| 6 | 2024-09-05 15:32:19 | test | t17 |  | auto analyze table all indexes, columns id, column1 with 256 buckets, 100 topn, 1 samplerate | 3000 | 2024-09-05 15:32:19 | 2024-09-05 15:32:19 | finished | null | 127.0.0.1:4000 | null |
| 7 | 2024-09-05 15:32:22 | test | t3 |  | auto analyze table all indexes, columns id, column1 with 256 buckets, 100 topn, 1 samplerate | 3000 | 2024-09-05 15:32:22 | 2024-09-05 15:32:22 | finished | null | 127.0.0.1:4000 | null |
| 8 | 2024-09-05 15:32:22 | test | t16 |  | auto analyze table all indexes, columns id, column1 with 256 buckets, 100 topn, 1 samplerate | 3000 | 2024-09-05 15:32:22 | 2024-09-05 15:32:22 | finished | null | 127.0.0.1:4000 | null |
| 9 | 2024-09-05 15:32:25 | test | t4 |  | auto analyze table all indexes, columns id, column1 with 256 buckets, 100 topn, 1 samplerate | 3000 | 2024-09-05 15:32:25 | 2024-09-05 15:32:25 | finished | null | 127.0.0.1:4000 | null |
| 10 | 2024-09-05 15:32:25 | test | t15 |  | auto analyze table all indexes, columns id, column1 with 256 buckets, 100 topn, 1 samplerate | 3000 | 2024-09-05 15:32:25 | 2024-09-05 15:32:25 | finished | null | 127.0.0.1:4000 | null |
| 11 | 2024-09-05 15:32:28 | test | t5 |  | auto analyze table all indexes, columns id, column1 with 256 buckets, 100 topn, 1 samplerate | 3000 | 2024-09-05 15:32:28 | 2024-09-05 15:32:28 | finished | null | 127.0.0.1:4000 | null |
| 12 | 2024-09-05 15:32:28 | test | t14 |  | auto analyze table all indexes, columns id, column1 with 256 buckets, 100 topn, 1 samplerate | 3000 | 2024-09-05 15:32:28 | 2024-09-05 15:32:28 | finished | null | 127.0.0.1:4000 | null |
| 13 | 2024-09-05 15:32:31 | test | t6 |  | auto analyze table all indexes, columns id, column1 with 256 buckets, 100 topn, 1 samplerate | 3000 | 2024-09-05 15:32:31 | 2024-09-05 15:32:31 | finished | null | 127.0.0.1:4000 | null |
| 14 | 2024-09-05 15:32:31 | test | t13 |  | auto analyze table all indexes, columns id, column1 with 256 buckets, 100 topn, 1 samplerate | 3000 | 2024-09-05 15:32:31 | 2024-09-05 15:32:31 | finished | null | 127.0.0.1:4000 | null |
| 15 | 2024-09-05 15:32:34 | test | t7 |  | auto analyze table all indexes, columns id, column1 with 256 buckets, 100 topn, 1 samplerate | 3000 | 2024-09-05 15:32:34 | 2024-09-05 15:32:34 | finished | null | 127.0.0.1:4000 | null |
| 16 | 2024-09-05 15:32:34 | test | t12 |  | auto analyze table all indexes, columns id, column1 with 256 buckets, 100 topn, 1 samplerate | 3000 | 2024-09-05 15:32:34 | 2024-09-05 15:32:34 | finished | null | 127.0.0.1:4000 | null |
| 17 | 2024-09-05 15:32:37 | test | t8 |  | auto analyze table all indexes, columns id, column1 with 256 buckets, 100 topn, 1 samplerate | 3000 | 2024-09-05 15:32:37 | 2024-09-05 15:32:37 | finished | null | 127.0.0.1:4000 | null |
| 18 | 2024-09-05 15:32:37 | test | t11 |  | auto analyze table all indexes, columns id, column1 with 256 buckets, 100 topn, 1 samplerate | 3000 | 2024-09-05 15:32:37 | 2024-09-05 15:32:37 | finished | null | 127.0.0.1:4000 | null |
| 19 | 2024-09-05 15:32:40 | test | t9 |  | auto analyze table all indexes, columns id, column1 with 256 buckets, 100 topn, 1 samplerate | 3000 | 2024-09-05 15:32:40 | 2024-09-05 15:32:40 | finished | null | 127.0.0.1:4000 | null |
| 20 | 2024-09-05 15:32:40 | test | t10 |  | auto analyze table all indexes, columns id, column1 with 256 buckets, 100 topn, 1 samplerate | 3000 | 2024-09-05 15:32:40 | 2024-09-05 15:32:40 | finished | null | 127.0.0.1:4000 | null |

As we can see, every time it will analyze 2 tables concurrently.

**Review by qw4990 — APPROVED** [review]

File: pkg/statistics/handle/autoanalyze/refresher/refresher_test.go [hunk 1]
```diff
@@ -351,6 +366,9 @@ func TestAnalyzeHighestPriorityTablesWithFailedAnalysis(t *testing.T) {
 	insertFailedJobForPartitionWithStartTime(tk, "test", "t1", "p0", startTime)
 
 	r.AnalyzeHighestPriorityTables()
+	require.Eventually(t, func() bool {
+		return len(r.GetRunningJobs()) == 0
+	}, 10*time.Second, 100*time.Millisecond)
```
Comment (hawkingrei) [inline pkg/statistics/handle/autoanalyze/refresher/refresher_test.go:371]: 10 seconds is excessive; most of these are set to 3 or 5 seconds

Comment (Rustin170506) [reply pkg/statistics/handle/autoanalyze/refresher/refresher_test.go]: Updated.

Comment (Rustin170506) [reply pkg/statistics/handle/autoanalyze/refresher/refresher_test.go]: I updated them to use `WaitAutoAnalyzeFinishedForTest`. We can avoid it at all.

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**